### PR TITLE
Keras 3 updates

### PR DESCRIPTION
**Pull request by divyashreepathihalli**
Uses tf.random.Generator instead of keras.backend.RandomGenerator
removed legacy_make_seed

Tests disabled - https://github.com/keras-team/keras-cv/issues/2167

all 3d preprocessing tests - has legacy baserandomlayer
rand_augment_test - need to investigate
random_shear_test - need to investigate
repeated_augmentation_test - need to investigate
resizing_test - has ragged tensor tests and is failing on jax and torch

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -8,105 +8,38 @@ permissions:
...
+            pip install -r requirements.txt --progress-bar off
+      - name: Build a binary wheel and a source tarball
+        run: |
+          export BUILD_WITH_CUSTOM_OPS=false
+          python pip_build.py
+      - name: Publish distribution to PyPI
+        if: startsWith(github.ref, 'refs/tags')
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          password: ${{ secrets.PYPI_API_TOKEN }}
```
Comment (mattdangerw) [inline .github/workflows/release.yml:45]: newline end of file

File: keras_cv/layers/preprocessing/base_image_augmentation_layer.py [hunk 2]
```diff
@@ -130,11 +130,13 @@ def augment_image(self, image, transformation):
     """
 
     def __init__(self, seed=None, **kwargs):
-        force_generator = kwargs.pop("force_generator", False)
-        self._random_generator = keras_backend.RandomGenerator(
-            seed=seed, force_generator=force_generator
-        )
+        if seed:
```
Comment (mattdangerw) [inline keras_cv/layers/preprocessing/base_image_augmentation_layer.py:133]: maybe better to do the explicit `if seed is not None` check here, avoid unexpected cases of people passing strange seed input and getting the global generator behavior

File: keras_cv/layers/preprocessing_3d/input_format_test.py [hunk 3]
```diff
@@ -15,98 +15,105 @@
...
 from absl.testing import parameterized
 
+from keras_cv.backend.config import keras_3
 from keras_cv.layers import preprocessing_3d
 from keras_cv.layers.preprocessing_3d import base_augmentation_layer_3d
 from keras_cv.tests.test_case import TestCase
 
-POINT_CLOUDS = base_augmentation_layer_3d.POINT_CLOUDS
-BOUNDING_BOXES = base_augmentation_layer_3d.BOUNDING_BOXES
+if not keras_3():
```
Comment (mattdangerw) [inline keras_cv/layers/preprocessing_3d/input_format_test.py:23]: is there a reason we can't just do this with the pytest annotation like elsewhere?

File: keras_cv/layers/serialization_test.py [hunk 4]
```diff
@@ -378,11 +379,30 @@ class SerializationTest(TestCase):
...
+                cv_layers.GlobalRandomScaling,
+                cv_layers.GlobalRandomTranslation,
+                cv_layers.GroupPointsByBoundingBoxes,
+                cv_layers.RandomCopyPaste,
+                cv_layers.RandomDropBox,
+                cv_layers.SwapBackground,
+                cv_layers.SqueezeAndExcite2D,  # TODO: Fails in Keras 3
+            ]
+            if layer_cls in skip_layers:
+                return
```
Comment (mattdangerw) [inline keras_cv/layers/serialization_test.py:400]: `self.skipTest("Not supported on Keras 3")` (this will show up in the test logs and make it obvious what is happening

File: requirements.txt [hunk 5]
```diff
@@ -1,14 +1,13 @@
...
-isort
-black
-pytest
-pycocotools
-keras-core
-build
-namex
+# Tensorflow.
+tf-nightly-cpu==2.16.0.dev20231109  # Pin a working nightly until rc0.
+keras-nightly==3.0.0.dev2023111903  # TODO recent nightly fails keras_cv/models/object_detection/retinanet/retinanet_test.py
```
Comment (mattdangerw) [inline requirements.txt:3]: typo with the path pasted here?

**Review by mattdangerw — APPROVED** [review]
Looks good! Don't have great visibility into what layers we need to disable for Keras 3 temporarily. But code overall looks clean!

File: requirements.txt (hunk 5 above)
Comment (sampathweb) [reply requirements.txt:3]: Thats how it's in PyPi - https://pypi.org/project/keras-nightly/3.0.0.dev2023111903/
The reason for it to look off is we may be saving down to Minute and TF only getting name down to the hour.

File: keras_cv/layers/serialization_test.py (hunk 4 above)
Comment (sampathweb) [reply keras_cv/layers/serialization_test.py]: That makes sense. Done

File: keras_cv/layers/preprocessing/base_image_augmentation_layer.py (hunk 2 above)
Comment (sampathweb) [reply keras_cv/layers/preprocessing/base_image_augmentation_layer.py]: That makes sense. Done

File: .github/workflows/release.yml (hunk 1 above)
Comment (sampathweb) [reply .github/workflows/release.yml]: Done

File: requirements.txt (hunk 5 above)
Comment (mattdangerw) [reply requirements.txt:3]: oh sorry my bad, the line was wrapping on `keras_cv/models/object_detection/retinanet/retinanet_test.py` so i didn't realize it was part of the comment for the failure

File: keras_cv/layers/preprocessing_3d/input_format_test.py (hunk 3 above)
Comment (sampathweb) [reply keras_cv/layers/preprocessing_3d/input_format_test.py:23]: The TEST_CONDITIONS list would also need to be moved inside the tests since those layers are also not compatible with Keras 3.  So left it as-is for now.  We can update it once we make them keras 3 compatible.
